### PR TITLE
modernize ember debug

### LINKED ISSUE #2457: Breaks on ember 5.2 w/ staticEmberSource: true
**Describe the bug**

Visit: 
- https://limber.glimdown.com
- or locally at http://localhost:4201/
- open the inspector
- observe: `Uncaught TypeError: _computed.readOnly is not a function`

**Screenshots**

image

Refreshing with the inspector open and "pause on exceptions" checked gives the actual problem code:

image

image

**Environment**
ember-source 5.2, all browsers

### PR DESCRIPTION
**Pull request by patricklx**
## Description
this modernizes ember debug to use modern js patterns and reduce usage of emberjs.
the only parts accessing emberjs is to do the actual debugging

fixes #2457, although there will not be tracked detection for static sources

### CODE REVIEW

**Comment by RobbieTheWagner** [comment]
Wow, this is a big one, but something we probably should have tackled a long time ago. Thank you for taking this on!

**Comment by patricklx** [comment]
this might also fix #2457

**Comment by RobbieTheWagner** [comment]
@patricklx mind fixing the conflicts? Hopefully I can review this soon.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -7692,8 +7702,8 @@ packages:
       - webpack
     dev: true
 
-  /ember-svg-jar@2.4.6:
-    resolution: {integrity: sha512-W+t6sRAiPhRl4imGKEMWsQDiCV8Nl2GRwzUpqZ1a+CZoW+2aCkF+kcngeDlHn5XzT16bQXg/QYXUvwtKJlSLfg==}
+  /ember-svg-jar@2.4.4:
```
Comment (RobbieTheWagner) [inline pnpm-lock.yaml:7705]: This seems to have been downgraded?

Comment (patricklx) [reply pnpm-lock.yaml:7705]: strange... great catch
i fix it

File: ember_debug/view-debug.js [hunk 2]
```diff
@@ -4,54 +4,56 @@ import RenderTree from 'ember-debug/libs/render-tree';
...
+
+      contextMenu() {
+        let { lastRightClicked } = this;
+        this.lastRightClicked = null;
+        this.inspectNearest(lastRightClicked);
+      },
+    };
+  }
 
   init() {
```
Comment (RobbieTheWagner) [inline ember_debug/view-debug.js:55]: Should this be `init` still or should we update it to `constructor`?

File: ember_debug/view-debug.js [hunk 3]
```diff
@@ -4,54 +4,56 @@ import RenderTree from 'ember-debug/libs/render-tree';
...
-  },
+export default class extends DebugPort {
+  get adapter() {
+    return this.namespace?.adapter;
+  }
+  get objectInspector() {
+    return this.namespace?.objectInspector;
+  }
+
+  static {
```
Comment (RobbieTheWagner) [inline ember_debug/view-debug.js:15]: Do we need static and prototype stuff here or could this just go on the class instance?

Comment (patricklx) [reply ember_debug/view-debug.js:15]: this is needed on prototype.
as `setupOrRemovePortListeners` accesses the `.messages` and `.portNamespace` in the constructor.
Which will happen before the sub class can set them up in their own constructor.
We could change this by either passing the messages and portNamespace as constrcutor param.
or by calling `setupOrRemovePortListeners` in the sub classes instead

Comment (patricklx) [reply ember_debug/view-debug.js:15]: actually, I can make it work the same way as Ember Core Object.
```
const obj = new this();
obj.init();
return obj;
````

File: ember_debug/view-debug.js (hunk 2 above)
Comment (patricklx) [reply ember_debug/view-debug.js:55]: i left init, because it gets overwritten in some parts in the tests. I didn't check which ones exactly

**Pull request merged**